src/Internal.AspNetCore.Sdk/build/Git.targets

```diff
@@ -0,0 +1,63 @@
+﻿<Project>
+
+  <PropertyGroup>
+    <GenerateCommitHashAttribute Condition="'$(GenerateCommitHashAttribute)'==''">true</GenerateCommitHashAttribute>
+    <GeneratedCommitHashAttributeFile Condition="'$(GeneratedCommitHashAttributeFile)'==''">$(IntermediateOutputPath)$(AssemblyName).CommitHash$(DefaultLanguageSourceExtension)</GeneratedCommitHashAttributeFile>
+  </PropertyGroup>
+
+  <Target Name="ResolveCommitHash" Condition="'$(CommitHash)'==''">
+    <GetGitCommitInfo WorkingDirectory="$(MSBuildProjectDirectory)" ContinueOnError="WarnAndContinue">
+      <Output TaskParameter="CommitHash" PropertyName="CommitHash" />
+    </GetGitCommitInfo>
+  </Target>
+
+<!--
+********************************************************************************************
+Target: GenerateCommitHashAttribute
+Generates an assembly attribute with commit hash
+********************************************************************************************
+-->
+  <Target Name="GenerateCommitHashAttribute"
+          BeforeTargets="CoreCompile"
+          DependsOnTargets="ResolveCommitHash;PrepareForBuild;GenerateIntermediateCommitHash;CoreGenerateCommitHashAttribute"
+          Condition="'$(GenerateCommitHashAttribute)'=='true' and '$(DesignTimeBuild)'!='true'" >
+    <Warning Text="Property 'CommitHash' was not set"
+             Condition="'$(CommitHash)'==''" />
+  </Target>
+
+  <Target Name="GenerateIntermediateCommitHash" Condition="'$(CommitHash)' != ''">
+    <PropertyGroup>
+      <!-- shorten to help avoid max path length issues -->
+      <IntermediateCommitHash>$(IntermediateOutputPath)$(CommitHash.Substring(0, 10)).commit</IntermediateCommitHash>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_OldCommitFiles Include="$(IntermediateOutputPath)*.commit" Exclude="$(IntermediateCommitHash)" />
+    </ItemGroup>
+
+    <Delete Files="@(_OldCommitFiles)" />
+    <Touch Files="$(IntermediateCommitHash)" AlwaysCreate="true" Condition="!Exists('$(IntermediateCommitHash)')" />
+  </Target>
+
+  <Target Name="CoreGenerateCommitHashAttribute"
```

Comment:
Super unrelated - is this the tabbing strategy we plan on using? I'd prefer we just tab because it means you don't have to go redo these if the value on the first line changes (doesn't necessarily apply here).